modules/endpoint-group/main.tf

```
@@ -1,5 +1,20 @@
 locals {
-  enabled = module.this.enabled
+  enabled                 = module.this.enabled
+  endpoint_configurations = try(var.config.endpoint_configuration, toset([]))
+  lb_names                = { for index, configuration in local.endpoint_configurations : index => try(configuration.endpoint_lb_name, null) }
+  eip_addresses           = { for index, configuration in local.endpoint_configurations : index => try(configuration.endpoint_eip_address, null) }
+}
+
+data "aws_lb" "lb" {
+  for_each = { for index, lb_name in local.lb_names : index => lb_name if lb_name != null }
```

Do not reference `index` in `for_each` as that defeats the purpose of making the entries independent of their place in the list.
```suggestion
  for_each = compact(values(local.lb_names))
```

---

@Nuru , yes, a single Global Accelerator Endpoint Group can have multiple endpoints

---

```suggestion
  for_each = local.lb_names 
```

---

Resolved